packages/webpack-extraction-plugin/src/nextPlugin.ts

```diff
@@ -0,0 +1,100 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import { GriffelCSSExtractionPlugin, GriffelCSSExtractionPluginOptions } from './GriffelCSSExtractionPlugin';
+import * as browserslist from 'browserslist';
+import { lazyPostCSS } from 'next/dist/build/webpack/config/blocks/css';
+import { getGlobalCssLoader } from 'next/dist/build/webpack/config/blocks/css/loaders';
+
+import type { NextConfig } from 'next';
+import type { Configuration, RuleSetRule } from 'webpack';
+import type { ConfigurationContext } from 'next/dist/build/webpack/config/utils';
+
+function getSupportedBrowsers(dir: string, isDevelopment: boolean) {
+  let browsers: string[] | undefined;
+  try {
+    browsers = browserslist.loadConfig({
+      path: dir,
+      env: isDevelopment ? 'development' : 'production',
+    });
+  } catch {
+    // Prevent error on build time
+  }
+
+  return browsers;
+}
+
+export const createGriffelNextExtractPlugin =
+  (pluginOptions: GriffelCSSExtractionPluginOptions = {}) =>
+  (nextConfig: NextConfig = {}) =>
+    Object.assign({}, nextConfig, {
+      webpack(config: Configuration & ConfigurationContext, options) {
+        const { dir, dev, isServer } = options;
+
+        if (!options.defaultLoaders) {
+          throw new Error(
+            'This plugin is not compatible with Next.js versions below 5.0.0 https://err.sh/next-plugins/upgrade',
+          );
+        }
+
+        // The @Griffel compiler must run on source code, which means it must be
+        // configured as the last loader in webpack so that it runs before any
+        // other transformation.
+
+        if (typeof nextConfig.webpack === 'function') {
+          // eslint-disable-next-line no-param-reassign
+          config = nextConfig.webpack(config, options);
+        }
+
+        if (dev) {
+          // not push @griffel/webpack-extraction-plugin if dev build
+          return config;
+        }
+
+        const cssRules = (
+          config.module?.rules?.find(
+            rule =>
+              typeof rule === 'object' &&
+              Array.isArray(rule.oneOf) &&
+              rule.oneOf.some(
+                ({ test }) => test instanceof RegExp && typeof test.test === 'function' && test.test('filename.css'),
+              ),
+          ) as RuleSetRule
+        )?.oneOf;
+
+        if (cssRules) {
+          config?.module?.rules?.unshift({
+            test: /\.(tsx|ts|js|jsx)$/,
+            exclude: /node_modules/,
```

Comment:
```suggestion
```

We need to run it on `node_modules` to extract styles that are coming with `@fluentui/react-components`.

Another option that I see is to make it configurable.